internal/wasm_utils/wasm_utils.go

```diff
@@ -0,0 +1,31 @@
+package wasm_utils
+
+import "syscall/js"
+
+// See https://stackoverflow.com/questions/68426700/how-to-wait-a-js-async-function-from-golang-wasm
+func Await(awaitable js.Value) ([]js.Value, []js.Value) {
+	then := make(chan []js.Value)
+	defer close(then)
+	thenFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		then <- args
+		return nil
+	})
+	defer thenFunc.Release()
+
+	catch := make(chan []js.Value)
+	defer close(catch)
+	catchFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		catch <- args
+		return nil
+	})
+	defer catchFunc.Release()
+
+	awaitable.Call("then", thenFunc).Call("catch", catchFunc)
+
+	select {
+	case result := <-then:
+		return result, nil
+	case err := <-catch:
+		return nil, err
+	}
+}
```

Comment:
This is some magic I don't fully understand, borrowed from [StackOverflow](https://stackoverflow.com/questions/68426700/how-to-wait-a-js-async-function-from-golang-wasm). But it works the same as `await` in JS!

Reply:
lol I'm both scared and thrilled!